json/src/main/java/io/airlift/json/JsonSubType.java

```diff
@@ -53,6 +55,20 @@ public class JsonSubType
 {
     private final Set<Module> modules;
 
+    public static boolean isJsonSubType(ObjectMapper objectMapper, Object instance)
+    {
+        if (instance != null) {
+            AnnotationIntrospector annotationIntrospector = objectMapper.getSerializationConfig().getAnnotationIntrospector();
+            if (annotationIntrospector != null) {
+                return annotationIntrospector.allIntrospectors()
+                        .stream()
+                        .flatMap(introspector -> (introspector instanceof TypeAddingIntrospector typeAddingIntrospector) ? Stream.of(typeAddingIntrospector) : Stream.empty())
```

Comment:
do we need flatMap? isn't filter+map enough?

Reply:
```
    public static boolean isJsonSubType(ObjectMapper objectMapper, Object instance)
    {
        if (instance == null) {
            return false;
        }
        AnnotationIntrospector annotationIntrospector = objectMapper.getSerializationConfig().getAnnotationIntrospector();
        if (annotationIntrospector == null) {
            return false;
        }

        return annotationIntrospector.allIntrospectors()
                .stream()
                .filter(TypeAddingIntrospector.class::isInstance)
                .map(TypeAddingIntrospector.class::cast)
                .anyMatch(typeAddingIntrospector -> typeAddingIntrospector.baseClass.isAssignableFrom(instance.getClass()));
    }
```

Reply:
It's for the cast to `TypeAddingIntrospector`

Reply:
I know. the filter+map does the same without creating streams and flattening them